src/contour/Config.h

```diff
@@ -144,263 +189,3020 @@ struct SimpleColorConfig
 };
 
 using ColorConfig = std::variant<SimpleColorConfig, DualColorConfig>;
-
-struct TerminalProfile
+enum class RenderingBackend
 {
-    vtpty::Process::ExecInfo shell;
-    bool maximized = false;
-    bool fullscreen = false;
-    bool showTitleBar = true;
-    bool sizeIndicatorOnResize = true;
-    bool mouseHideWhileTyping = true;
-    vtbackend::RefreshRate refreshRate = { 0.0 }; // 0=auto
-    vtbackend::LineOffset copyLastMarkRangeOffset = vtbackend::LineOffset(0);
-
-    std::string wmClass;
-
-    vtbackend::PageSize terminalSize = { vtbackend::LineCount(10), vtbackend::ColumnCount(40) };
-    vtbackend::VTType terminalId = vtbackend::VTType::VT525;
-
-    vtbackend::MaxHistoryLineCount maxHistoryLineCount;
-    vtbackend::LineCount historyScrollMultiplier = vtbackend::LineCount(3);
-    ScrollBarPosition scrollbarPosition = ScrollBarPosition::Right;
-    vtbackend::StatusDisplayPosition statusDisplayPosition = vtbackend::StatusDisplayPosition::Bottom;
-    bool syncWindowTitleWithHostWritableStatusDisplay = false;
-    bool hideScrollbarInAltScreen = true;
-    bool optionKeyAsAlt = false;
+    Default,
+    Software,
+    OpenGL,
+};
 
-    bool autoScrollOnUpdate;
+template <typename T>
+struct ConfigEntry
+{
+    using value_type = T;
 
-    vtrasterizer::FontDescriptions fonts;
+    constexpr ConfigEntry(string d): _value {}, _documentation { std::move(d) } {}
 
-    struct
+    constexpr ConfigEntry(T&& in, string d): _value { std::forward<T>(in) }, _documentation { std::move(d) }
```

Comment:
```suggestion
    constexpr ConfigEntry(T&& in, std::string d): _value { std::forward<T>(in) }, _documentation { std::move(d) }
```

I'm not going to comment that now everywhere below. This is most likely because you imported std::string into the current namespace. The problem I see here is, that this is a public header, and that should not have happened - IMHO at least :)